BladeportBinaryTreeManager.Web/wwwroot/app/src/app/user/user.service.ts

```diff
@@ -1,17 +1,30 @@
 import { Injectable } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
 import { Observable } from 'rxjs';
+import { map } from "rxjs/operators";
+import { UserObj, UserObjAdapter } from './user.model';
+import { isNumeric } from 'jquery';
 
 @Injectable()
 export class UserService {
     
     private _server = 'https://localhost:44358/'
+    //private _server = 'http://localhost:5000/'
     private _getUsersUrl = 'api/user/users';
 
-    constructor(private _http: HttpClient) {}
+    constructor(private _http: HttpClient, private _adapter: UserObjAdapter) {}
 
     getUserList(): Observable<any> {
-        return this._http.get(this._server + this._getUsersUrl);            
+        return this._http.get(this._server + this._getUsersUrl);
+    }
+    
+    getUsers(): Observable<UserObj[]> {
+        const url = this._server + this._getUsersUrl;
+
+        // Adapt each item in the raw data array
+        return this._http.get(url).pipe(
+            map((data: any[]) => data.map((item: any) => this._adapter.adapt(item)))
```

Comment:
create a model in typescript that matches your API response so that the UserService can fetch the data from the API using the exact data types from source.

e.g.

.NET:

public class User {
     public int UserId { get; set; }
    public string UserName {get; set;}
    public string LastName {get; set;}
}

Typescript:

export interface IUser {
    UserId: number;
    UserName: string;
    LastName: string;
}